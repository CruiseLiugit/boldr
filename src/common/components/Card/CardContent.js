/* @flow */
import React from 'react';
import Heading from '../Heading';

type Props = {
  children: ReactElement,
  title: String,
  footerMessage: String,
};
const CardContent = (props: Props) => {
  return (
    <div className="boldr-card__content">
      <Heading classname="boldr-card__title" size={ 3 }>
        { props.title }
      </Heading>
      { props.children }
      { props.footerMessage ? (
        <div className="boldr-card__footer-message">
          { props.footerMessage }
        </div>) : null
      }
    </div>
  );
};

export default CardContent;
