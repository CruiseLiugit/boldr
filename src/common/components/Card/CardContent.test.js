import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CardContent from './CardContent';

it('<CardContent />, should render', () => {
  const wrapper = shallow(<CardContent>Hey</CardContent>);
  const wrapperInner = wrapper.find('.boldr-card__content');
  expect(wrapperInner.is('.boldr-card__content')).toBe(true);
});

it('renders snapshot', () => {
  const wrapper = shallow(<CardContent>Hey</CardContent>);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
