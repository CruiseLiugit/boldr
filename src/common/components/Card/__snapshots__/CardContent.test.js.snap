exports[`test renders snapshot 1`] = `
<div
  className="boldr-card__content">
  <Component
    classname="boldr-card__title"
    size={3} />
  Hey
</div>
`;
